src/pages/trade/model/positions.ts

```diff
@@ -107,39 +107,74 @@ class PositionsStore {
     try {
       this.setLoading(true);
 
-      // Fetching latest position data as the planner request requires current reserves + pair
-      const promises = positions.map(({ id }) =>
-        penumbra.service(DexService).liquidityPositionById({ positionId: id }),
-      );
-      const latestPositionData = await Promise.all(promises);
-
-      // a position can be closed remotely, but not yet closed locally
-      // in this case we need to generate an nft to be able to withdraw
-      // it and include it in the planner request
-      const positionIdsToClose = positions
-        .map(({ id, position }, i) => ({
+      // Our goal here is to withdraw all the closed position in this subaccount.
+      // Problem:
+      // 1. Auto-closing position switch to `Closed` without user input
+      // 2. We are building a list of positions to withdraw based on the latest on-chain state.
+      // 3. This list might contain position for which we do not yet own an associated closed LP NFT.
+      // Solution:
+      // 1. Track auto-closing positions that we are going to withdraw
+      // 2. Inspect our balance to check if we have an associated opened NFT
+      // 3. If that's the case, add a `PositionClose` action to the TPR.
+      // Later, we can improve the general explorer data flow, for now we should just ship this and
+      // make this flow work. This is, at the moment, the **ONLY** thing that matters.
+
+      // First track all the positions we want to withdraw.
+      const positionWithdraws = positions
+        .filter(({ position }) => {
+          return position.state?.state === PositionState_PositionStateEnum.CLOSED;
+        })
+        .map(({ id, position }) => ({
           positionId: id,
-          prevState: position.state,
-          nextState: latestPositionData[i]?.data?.state,
-        }))
+          tradingPair: position.phi?.pair,
+          reserves: position.reserves,
+        }));
+
+      // Return early if there's no work to do.
+      if (!positionWithdraws.length) {
+        return;
+      }
+
+      // TODO(jason): not sure if this is duplicating code, feel free to move it out somewhere less disruptive.
+      async function asyncIterableToArray<T>(asyncIterable: AsyncIterable<T>): Promise<T[]> {
+        const array: T[] = [];
+        for await (const item of asyncIterable) {
+          array.push(item);
+        }
+        return array;
+      }
+
+      // Query the balance, ignoring the subaccount index for now.
+      const balances = await asyncIterableToArray(penumbra.service(ViewService).balances({}));
```

Comment:
Here you can do:
```suggestion
      const balances = await Array.fromAsync(penumbra.service(ViewService).balances({}));
```

Reply:
It seems we have a `useBalances` hook in `src/shared/api/balances.ts`. I feel like we should probably reuse that, that one also accept the account index to fetch the balances by the selected sub account.